### ISSUE TITLE
Nightly crashing on startup

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#13816]
### Describe your problem!

Several users report, that nightly 2023.01.03 is crashing on startup.
I can reproduce on my device.

Seems to be a permission issue for existing installations.

### How to reproduce?

Start c:geo after upgrading to 2023.01.03 nightly.

### Actual result after these steps?

```
2023-01-03 23:13:50.583 13544-13544 Andro...time pid-13544        E  FATAL EXCEPTION: main
                                                                     Process: cgeo.geocaching, PID: 13544
                                                                     java.lang.RuntimeException: Unable to resume activity {cgeo.geocaching/cgeo.geocaching.MainActivity}: java.lang.SecurityException: Permission android.permission.READ_EXTERNAL_STORAGE denied for package cgeo.geocaching
                                                                     	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:5365)
                                                                     	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:5431)
                                                                     	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:54)
                                                                     	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
                                                                     	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
                                                                     	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
                                                                     	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2571)
                                                                     	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                     	at android.os.Looper.loopOnce(Looper.java:226)
                                                                     	at android.os.Looper.loop(Looper.java:313)
                                                                     	at android.app.ActivityThread.main(ActivityThread.java:8741)
                                                                     	at java.lang.reflect.Method.invoke(Native Method)
                                                                     	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
                                                                     	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
                                                                     Caused by: java.lang.SecurityException: Permission android.permission.READ_EXTERNAL_STORAGE denied for package cgeo.geocaching
                                                                     	at android.os.Parcel.createExceptionOrNull(Parcel.java:3023)
                                                                     	at android.os.Parcel.createException(Parcel.java:3007)
                                                                     	at android.os.Parcel.readException(Parcel.java:2990)
                                                                     	at android.os.Parcel.readException(Parcel.java:2932)
                                                                     	at android.app.IWallpaperManager$Stub$Proxy.getWallpaperWithFeature(IWallpaperManager.java:1996)
                                                                     	at android.app.WallpaperManager$Globals.getCurrentWallpaperLocked(WallpaperManager.java:1073)
                                                                     	at android.app.WallpaperManager$Globals.peekWallpaperBitmap(WallpaperManager.java:936)
                                                                     	at android.app.WallpaperManager$Globals.peekWallpaperBitmap(WallpaperManager.java:863)
                                                                     	at android.app.WallpaperManager.getDrawable(WallpaperManager.java:1380)
                                                                     	at cgeo.geocaching.MainActivity.onResume(MainActivity.java:459)
                                                                     	at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1531)
                                                                     	at android.app.Activity.performResume(Activity.java:8721)
                                                                     	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:5338)
                                                                     	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:5431) 
                                                                     	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:54) 
                                                                     	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45) 
                                                                     	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176) 
                                                                     	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97) 
                                                                     	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2571) 
                                                                     	at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                     	at android.os.Looper.loopOnce(Looper.java:226) 
                                                                     	at android.os.Looper.loop(Looper.java:313) 
                                                                     	at android.app.ActivityThread.main(ActivityThread.java:8741) 
                                                                     	at java.lang.reflect.Method.invoke(Native Method) 
                                                                     	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571) 
                                                                     	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067) 
                                                                     Caused by: android.os.RemoteException: Remote stack trace:
                                                                     	at android.os.storage.StorageManager.checkPermissionAndAppOp(StorageManager.java:1915)
                                                                     	at android.os.storage.StorageManager.checkPermissionAndAppOp(StorageManager.java:1892)
                                                                     	at android.os.storage.StorageManager.checkPermissionAndAppOp(StorageManager.java:1959)
                                                                     	at android.os.storage.StorageManager.checkExternalStoragePermissionAndAppOp(StorageManager.java:2075)
                                                                     	at android.os.storage.StorageManager.checkPermissionReadImages(StorageManager.java:2041)
2023-01-03 23:13:50.700  1408-1453  Windo...ager pid-1408         E  win=Window{8743936 u0 Splash Screen cgeo.geocaching EXITING} destroySurfaces: appStopped=false cleanupOnResume=false win.mWindowRemovalAllowed=true win.mRemoveOnExit=true win.mViewVisibility=0 caller=com.android.server.wm.WindowState.onExitAnimationDone:5917 com.android.server.wm.ActivityRecord$$ExternalSyntheticLambda11.accept:2 java.util.ArrayList.forEach:1262 com.android.server.wm.ActivityRecord.onAnimationFinished:8440 com.android.server.wm.ActivityRecord.postApplyAnimation:6146 com.android.server.wm.ActivityRecord.commitVisibility:6118 com.android.server.wm.ActivityRecord.commitVisibility:6122 
2023-01-03 23:13:53.343 13704-13704 Andro...time pid-13704        E  FATAL EXCEPTION: main
                                                                     Process: cgeo.geocaching, PID: 13704
                                                                     java.lang.RuntimeException: Unable to resume activity {cgeo.geocaching/cgeo.geocaching.MainActivity}: java.lang.SecurityException: Permission android.permission.READ_EXTERNAL_STORAGE denied for package cgeo.geocaching
                                                                     	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:5365)
                                                                     	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:5431)
                                                                     	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:54)
                                                                     	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
                                                                     	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
                                                                     	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
                                                                     	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2571)
                                                                     	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                     	at android.os.Looper.loopOnce(Looper.java:226)
                                                                     	at android.os.Looper.loop(Looper.java:313)
                                                                     	at android.app.ActivityThread.main(ActivityThread.java:8741)
                                                                     	at java.lang.reflect.Method.invoke(Native Method)
                                                                     	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
                                                                     	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
                                                                     Caused by: java.lang.SecurityException: Permission android.permission.READ_EXTERNAL_STORAGE denied for package cgeo.geocaching
                                                                     	at android.os.Parcel.createExceptionOrNull(Parcel.java:3023)
                                                                     	at android.os.Parcel.createException(Parcel.java:3007)
                                                                     	at android.os.Parcel.readException(Parcel.java:2990)
                                                                     	at android.os.Parcel.readException(Parcel.java:2932)
                                                                     	at android.app.IWallpaperManager$Stub$Proxy.getWallpaperWithFeature(IWallpaperManager.java:1996)
                                                                     	at android.app.WallpaperManager$Globals.getCurrentWallpaperLocked(WallpaperManager.java:1073)
                                                                     	at android.app.WallpaperManager$Globals.peekWallpaperBitmap(WallpaperManager.java:936)
                                                                     	at android.app.WallpaperManager$Globals.peekWallpaperBitmap(WallpaperManager.java:863)
                                                                     	at android.app.WallpaperManager.getDrawable(WallpaperManager.java:1380)
                                                                     	at cgeo.geocaching.MainActivity.onResume(MainActivity.java:459)
                                                                     	at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1531)
                                                                     	at android.app.Activity.performResume(Activity.java:8721)
                                                                     	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:5338)
                                                                     	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:5431) 
                                                                     	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:54) 
                                                                     	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45) 
                                                                     	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176) 
                                                                     	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97) 
                                                                     	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2571) 
                                                                     	at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                     	at android.os.Looper.loopOnce(Looper.java:226) 
                                                                     	at android.os.Looper.loop(Looper.java:313) 
                                                                     	at android.app.ActivityThread.main(ActivityThread.java:8741) 
                                                                     	at java.lang.reflect.Method.invoke(Native Method) 
                                                                     	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571) 
                                                                     	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067) 
                                                                     Caused by: android.os.RemoteException: Remote stack trace:
                                                                     	at android.os.storage.StorageManager.checkPermissionAndAppOp(StorageManager.java:1915)
                                                                     	at android.os.storage.StorageManager.checkPermissionAndAppOp(StorageManager.java:1892)
                                                                     	at android.os.storage.StorageManager.checkPermissionAndAppOp(StorageManager.java:1959)
                                                                     	at android.os.storage.StorageManager.checkExternalStoragePermissionAndAppOp(StorageManager.java:2075)
                                                                     	at android.os.storage.StorageManager.checkPermissionReadImages(StorageManager.java:2041)
```

### Reproducible

Yes

### c:geo Version

2023.01.03-NB

**Comment by Lineflyer** [comment]
Addition as I cannot reproduce on API32 emulator:
My device is running Android 13. Asked the user on support about their versions.

**Comment by eddiemuc** [comment]
From the stack trace I assume that we need Read-external-storage permission for those users who use a wallpaper inside c:geo. Interesting.

Is this use case important enough to request this permission from all users?

As a quick fix we could deactivate wallpaper support.

**Comment by eddiemuc** [comment]
https://issuetracker.google.com/issues/237124750

**Comment by Lineflyer** [comment]
WTF.....how can it be, that this is not foreseen to use this wallpaper feature.

Indeed I have a wallpaper set on my device.
From my understanding: If we remove that feature we will probably never be able to add it again, as there will be devices out there still not supporting it on their versions although Google (if they do after all) fixes it...

While wallpaper support is far away from being an essential feature, its still kind of a "trademark" of c:geo which we revived with the wallpaper feature. It would be boring to lose it.

**Comment by Lineflyer** [comment]
BTW:
Can`t we ask for read storage permission on runtime as soon as a user activates the wallpaper feature?

**Comment by eddiemuc** [comment]
> BTW:
> Can`t we ask for read storage permission on runtime as soon as a user activates the wallpaper feature?

Should be possible

**Comment by eddiemuc** [comment]
Fixed with PR #13817
